bindings/python/test/estimator/test_od_least_squares_solver.py

```diff
@@ -0,0 +1,120 @@
+# Apache License 2.0
+
+import pytest
+
+import pandas as pd
+
+from ostk.physics import Environment
+from ostk.physics.time import Instant
+from ostk.physics.coordinate import Frame
+from ostk.physics.environment.object.celestial import Earth
+
+from ostk.astrodynamics.solver import LeastSquaresSolver
+from ostk.astrodynamics.trajectory import State
+from ostk.astrodynamics.trajectory import Orbit
+from ostk.astrodynamics.trajectory import Propagator
+from ostk.astrodynamics.trajectory.state import NumericalSolver
+from ostk.astrodynamics.trajectory.state import CoordinateSubset
+from ostk.astrodynamics.trajectory.state.coordinate_subset import CartesianPosition
+from ostk.astrodynamics.trajectory.state.coordinate_subset import CartesianVelocity
+from ostk.astrodynamics.estimator import ODLeastSquaresSolver
+from ostk.astrodynamics.dataframe import generate_states_from_dataframe
+
+
+@pytest.fixture
+def environment() -> Environment:
+    return Environment(central_celestial_object=Earth.EGM96(10, 10), objects=[])
+
+
+@pytest.fixture
+def numerical_solver() -> NumericalSolver:
+    return NumericalSolver.default()
+
+
+@pytest.fixture
+def least_squares_solver() -> LeastSquaresSolver:
+    return LeastSquaresSolver.default()
+
+
+@pytest.fixture
+def od_solver(
+    environment: Environment,
+    numerical_solver: NumericalSolver,
+    least_squares_solver: LeastSquaresSolver,
+) -> ODLeastSquaresSolver:
+    return ODLeastSquaresSolver(
+        environment=environment,
+        numerical_solver=numerical_solver,
+        solver=least_squares_solver,
+    )
+
+
+@pytest.fixture
+def coordinate_subsets() -> list[CoordinateSubset]:
+    return [
+        CartesianPosition.default(),
+        CartesianVelocity.default(),
+    ]
+
+
+@pytest.fixture
+def initial_guess_state(
+    reference_states: list[State],
+) -> State:
+    return reference_states[0]
+
+
+@pytest.fixture
+def reference_states() -> list[State]:
+    return generate_states_from_dataframe(
+        pd.read_csv("/app/test/OpenSpaceToolkit/Astrodynamics/Estimator/gnss_data.csv"),
```

Comment:
hmmm should we use the gps data for the python tests? I think we should just keep the actual unit tests in cpp and not in the bindings no?

Reply:
It was easier than creating fake states lol